src/reporting/clients/ODataClient.ts

```diff
@@ -0,0 +1,119 @@
+        "odataItem",
+        "odata item was invalid when calling getODataReportEntityPage.",
+      );
+    }
+    const url = `${this.basePath}/odata/${encodeURIComponent(reportId)}/${odataItem.url}?sequence=${encodeURIComponent(sequence)}`;
+    const requestOptions: RequestInit = this.createRequest("GET", accessToken);
+    return this.fetchJSON(url, requestOptions);
+  }
+
+  public async getODataReportEntities(accessToken: AccessToken, reportId: string, odataItem: ODataItem): Promise<Array<ODataEntityValue>> {
+    const segments = odataItem.url.split("/");  // region, manifestId, entityType
+    if (segments.length !== 3) {
+      throw new RequiredError(
+        "odataItem",
+        "odata item was invalid when calling getODataReportEntities.",
+      );
+    }
+    const reportData: Array<ODataEntityValue> = [];
+    const oDataReportEntitiesIt = this.getODataReportEntitiesIterator(accessToken, reportId, odataItem);
+    for await(const oDataReportEntity of oDataReportEntitiesIt) {
+      reportData.push(oDataReportEntity);
+    }
+    return reportData;
+  }
+
+  public getODataReportEntitiesIterator(accessToken: AccessToken, reportId: string, odataItem: ODataItem): EntityListIterator<ODataEntityValue> {
+    const segments = odataItem.url.split("/");  // region, manifestId, entityType
+    if (segments.length !== 3) {
+      throw new RequiredError(
+        "odataItem",
+        "odata item was invalid when calling getODataReportEntitiesIterator.",
+      );
+    }
+    const url = `${this.basePath}/odata/${encodeURIComponent(reportId)}/${odataItem.url}`;
+    const request = this.createRequest("GET", accessToken);
+    return new EntityListIteratorImpl(async () => getEntityCollectionPage<ODataEntityValue>(
+      url,
+      async (nextUrl: string): Promise<Collection<ODataEntityValue>> => {
+        const response: ODataEntityResponse = await this.fetchJSON<ODataEntityResponse>(nextUrl, request);
+        const link: PagedResponseLinks = {
+          self: {
+            href: nextUrl,
+          },
+        };
+        if(response["@odata.nextLink"]) {
+          link.next = {
+            href: response["@odata.nextLink"],
+          };
+        }
+        return {
+          values: response.value,
+          // eslint-disable-next-line @typescript-eslint/naming-convention
+          _links: link,
+        };
+      }));
+  }
+
+  public async getODataReportMetadata(accessToken: AccessToken, reportId: string): Promise<DisplayTable[]> {
+    const url = `${this.basePath}/odata/${encodeURIComponent(reportId)}/$metadata`;
+    const requestOptions: RequestInit = this.createRequest("GET", accessToken);
+    const response = await this.fetchData(url, requestOptions);
+
+    const options = {
+      ignoreAttributes: false,
+      attributeNamePrefix : "",
+    };
+    const parser = new XMLParser(options);
+    const parsedXML = parser.parse(await response.text());
+    if (!parsedXML["edmx:Edmx"]["edmx:DataServices"].Schema[0]) {
```

Comment:
You should filter `Schema` by its namespace, not by its index.

Reply:
Changed to check if schema is an array since if there is only one element (default namespace) the parser will return an object and not an array. Also added a check for schema element, added test for this.

Reply:
If there is at least 1 table, there will be 2 Schemas - 1 Default and 1 non default. Client should handle any potential changes of how we distribute EntityTypes across schemas. EntitySet attribute Name is the actual table name. EntitySet attribute EntityType points to the EntityType definition. EntitySet's attribute EntityType value is in the format of `{Schema Namespace}.{EntityType name}`.

Reply:
applied this change